posts/indu-plp-part3/plp-part3.qmd

```diff
@@ -0,0 +1,147 @@
+
+Temporal context was also missing, as the features extracted were basic binary indicators that did not reflect the timing or frequency of clinical events. The one-year prediction window might have been too narrow for diabetes to develop meaningfully after a hypertension diagnosis. 
+
+Additionally, many patients had very limited observational data before the index date, reducing the reliability of the features. Altogether, the problem lay not with the modeling itself but with the mismatch between the available data and the clinical nuances of the question being asked.
+
+## Lessons Learned
+
+This project highlighted that Julia is well-suited for observational health research workflows. Its strong data ecosystem (`DataFrames.jl`, `DuckDB.jl`, `CSV.jl`), rich machine learning interfaces (`MLJ.jl`), and increasing support from domain-specific packages like `OHDSICohortExpressions.jl` make it a compelling environment for building transparent, customizable pipelines. Unlike black-box tools, Julia allows full control over every stage-from data access to model tuning-encouraging deeper understanding and reproducibility.
+
+From this experience, several lessons became evident:
+
+- Even with a standardized data model like OMOP CDM, **cohort construction requires clinical reasoning and thorough validation**. Poorly defined cohorts can introduce significant noise and reduce the quality of the learning task.
+- **Simple features such as binary flags often fail** to capture the complex temporal and contextual dynamics in longitudinal health data.
+- **Model performance depends on the entire pipeline**, not just the algorithm. Label noise, data completeness, and preprocessing choices all significantly impact the results.
+- **Reproducibility is essential**-having modular code and clearly defined steps ensures that results can be validated, shared, and extended by others in the community.
+
+
+## Key Challenges Faced
+
+Throughout the process, we encountered several technical and conceptual challenges:
+
+The most critical issue lay in the quality and structure of the data itself. Many patients had sparse or short observation periods, which meant that only limited clinical history was available before the cohort entry date. 
+
+This directly affected the utility of the extracted features. On the modeling side, integrating with `MLJ.jl` required careful setup, especially when handling missing values, categorical encodings, and class imbalance.
+
+<br>
+<center>
+  ![](./mlj.png)
+
+  MLJ.jl
+</center>
+<br>
+
+Feature engineering was another bottleneck, since most features were binary flags or simple counts that did not capture clinical nuance or temporal dynamics. Preprocessing steps like normalization and imputation also needed fine-tuning and often had to be manually adjusted per model. 
+
+## Next Steps for the Pipeline
```

Comment:
Can you add more of your own thoughts and perspectives here? Maybe about what personally excites you for future explorations and work?